src/data/repositories/UserD2Repository.ts

```diff
@@ -35,6 +68,23 @@ const userFields = {
         username: true,
         userRoles: { id: true, name: true, authorities: true },
     },
+    settings: {
+        keyUiLocale: true,
+    },
+} as const;
+
+const constantFields = {
+    id: true,
+    code: true,
+    description: true,
 } as const;
 
-type D2User = MetadataPick<{ users: { fields: typeof userFields } }>["users"][number];
+const constantDescriptionCodec = Codec.interface({ administratorGroups: array(string) });
+const constantsErrMsg =
+    "Unable to retrieve if the user is Admin (can select all languages). The TALLY_SHEETS_STORAGE description is not a valid JSON object.";
```

Comment:
Use the code from a single source of truth so it's not even possible to make typos! (we'll be creating d2-metadata.ts or similar)